// Copyright (c) 2023 Boston Dynamics, Inc.  All rights reserved.
//
// Downloading, reproducing, distributing or otherwise using the SDK Software
// is subject to the terms and conditions of the Boston Dynamics Software
// Development Kit License (20191101-BDSDK-SL).

syntax = "proto3";

package bosdyn.api.spot_cam;

option java_outer_classname = "VersionProto";

import "bosdyn/api/header.proto";
import "bosdyn/api/robot_id.proto";

// Request the software version running on the SpotCam.
message GetSoftwareVersionRequest {
    // Common request header.
    bosdyn.api.RequestHeader header = 1;
}

// Provide the SpotCam's software release version.
message GetSoftwareVersionResponse {
    // Common response header.
    bosdyn.api.ResponseHeader header = 1;
    // Version of the software currently running on the SpotCam.
    bosdyn.api.SoftwareVersion version = 2;
    // Extra detail about the version of software running on spotcam.
    // May contain metadata about build dates and configuration.
    string detail = 3;
}
